core-library/src/main/java/org/silverpeas/core/index/search/SimpleSearchEngine.java

```diff
@@ -173,17 +174,27 @@ private boolean isMatchingIndexEntryAvailable(MatchingIndexEntry mie, String use
           || "Node".equals(objectType);
     }
 
-    // verify rights according to 'kmelia' rights
+    // verify rights according to component specific rights
     String componentId = mie.getComponent();
-    if (componentId.startsWith("kmelia")) {
-      return authorization
-          .isObjectAvailable(componentId, userId, mie.getObjectId(), mie.getObjectType());
+    ComponentAuthorization auth = getAuthorization(componentId, authorizations);
+    if (auth != null) {
+      return auth.isObjectAvailable(componentId, userId, mie.getObjectId(), mie.getObjectType());
     }
 
     // verify rights onto others items type
     return isOtherItemAvailable(mie, userId, allowedComponentIds);
   }
 
+  private ComponentAuthorization getAuthorization(String componentId,
+      HashMap<String, ComponentAuthorization> auths) {
+    if (componentId.startsWith("kmelia")) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "kmelia" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)